### PR TITLE
Fix/normalizer produces, list was empty. is now populated and links to object page

### PR DESCRIPTION
**Pull request by underdarknl**
### Changes
Our normalizer detail pages had a header listing all produced ooi types.
The normalizer schema's contain the produces ooi_types, however, they didn't make it to the actual template.
This is now fixed, along with some other minor issues regarding the boefje detail page.

### Demo
image

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have made corresponding changes to the documentation, if necessary.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: rocky/katalogus/client.py [hunk 1]
```diff
@@ -99,8 +103,8 @@ def get_plugin_schema(self, plugin_id) -> Optional[Dict]:
         try:
             Draft202012Validator.check_schema(schema)
             return schema
-        except SchemaError:
-            logger.warning("Invalid schema found for plugin %s", plugin_id)
+        except SchemaError as error:
+            logger.warning("Invalid schema found for plugin %s, %s", plugin_id, error)
```
Comment (ammar92) [inline rocky/katalogus/client.py:107]: If you intended to only show the error message in the exception (which generally is indeed casting an exception to a string) then this is the right way. If you wanted to show the traceback, it's better to call a logger method (such as `warning` or `info`) with the parameter `exc_info=True` (or `exc_info=error`)

File: rocky/katalogus/templates/normalizer_detail.html [hunk 2]
```diff
@@ -62,7 +64,14 @@ <h2>{% translate "Produces" %}</h2>
...
                 </p>
                 <p>
-                    {% for ooi in plugin.produces %}{{ ooi }}{% endfor %}
+                    {% if plugin.produces %}
+                        {% for ooi_type in plugin.produces %}
+                            <a href="{% url "ooi_list" organization_code=organization.code %}?ooi_type={{ ooi_type|urlencode:"" }}">{{ ooi_type }}</a>
+                            {% if not forloop.last %}-{% endif %}
+                        {% endfor %}
+                    {% else %}
+                        No known produces OOI's listed. N.B. This list might not be complete.
```
Comment (ammar92) [inline rocky/katalogus/templates/normalizer_detail.html:73]:
```suggestion
                        No known produces OOIs listed. N.B. This list might not be complete.
```

Comment (ammar92) [reply rocky/katalogus/templates/normalizer_detail.html:73]: "No known produces OOIs listed." was a bit hard to understand, but I don't have an alternative (yet)

**Review by ammar92 — APPROVED** [review]

**Comment by stephanie0x00** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
On the Katalogus page the following items are now shown (again) after clicking the details button for boefjes and normalizers: 
- the Consumes and Produces-attributes on boefje and normalizers
- when clicking the details on a boefje it also shows the tasks that have run (if available), including the object list. 

### What doesn't work:
For normalizers no tasks are shown, even though the normalizer has run. I believe this is picked up in a different ticket? 

### Bug or feature?:
n/a

File: rocky/katalogus/client.py (hunk 1 above)
Comment (underdarknl) [reply rocky/katalogus/client.py:107]: At the moment, the schema's all seem to be invalid, and filling up the logs, just letting the user know roughly what went wrong in the schema is enough.

**Comment by underdarknl** [comment]
> For normalizers no tasks are shown, even though the normalizer has run. I believe this is picked up in a different ticket?

Yes, Thursday, another PR was merged that should have solved that issue.

File: rocky/katalogus/client.py [hunk 3]
```diff
@@ -195,10 +202,13 @@ def parse_normalizer(normalizer: Dict) -> Normalizer:
     name = normalizer["id"].replace("_", " ").replace("kat ", "").title()
 
     consumes = set(normalizer["consumes"])
+    consumes.add(f"normalizer/{name.lower()}")
```
Comment (dekkers) [inline rocky/katalogus/client.py:205]: This should be

```suggestion
    consumes.add(f"normalizer/{normalizer['id'}")
```

to match https://github.com/minvws/nl-kat-coordination/blob/main/boefjes/boefjes/plugins/models.py#L69

But given that the katalogus adds it when reading the normalizer definition, shouldn't this already be included?

Comment (underdarknl) [reply rocky/katalogus/client.py:205]: There are at least some issues where the mime-type for a normaliser (as listed on this page) looses its underscores where it shouldn't.

**Review by dekkers — DISMISSED** [review]

**Review by dekkers — APPROVED** [review]

**Pull request merged**